### PR TITLE
Include <x86intrin.h> only if defined(__x86_64__)

### PR DESCRIPTION
**Pull request by cary-ilm**
Hopefully address #1105?

Signed-off-by: Cary Phillips <cary@ilm.com>

### CODE REVIEW

File: src/lib/OpenEXRCore/internal_coding.h [hunk 1]
```diff
@@ -24,8 +24,8 @@
 #    include <half.h>
 #endif
 
-#if defined(__has_include)
-#    if __has_include(<x86intrin.h>)
+#if defined(__has_include) 
+#    if __has_include(<x86intrin.h>) && defined(__x86_64__)
```
Comment (carlocab) [inline src/lib/OpenEXRCore/internal_coding.h]:
```suggestion
#if defined(__has_include) && defined(__x86_64__)
#    if __has_include(<x86intrin.h>)
```
macOS ARM64 doesn't seem to have `intrin.h`. This is also the same fix from AcademySoftwareFoundation/Imath#184.

Comment (meshula) [reply src/lib/OpenEXRCore/internal_coding.h]: @carlocab's suggestion is correct. Also, we need stdbool.h as per 

diff --git a/src/lib/OpenEXRCore/internal_zip.c b/src/lib/OpenEXRCore/internal_zip.c
index 0192ac2a..dc32794b 100644
--- a/src/lib/OpenEXRCore/internal_zip.c
+++ b/src/lib/OpenEXRCore/internal_zip.c
@@ -11,6 +11,7 @@
 
 #include <limits.h>
 #include <stdlib.h>
+#include <stdbool.h>
 #include <string.h>
 #include <zlib.h>

**Comment by cary-ilm** [comment]
I pushed a commit that fixes the include as well as the stdbool.h. 

But @meshula, what about the suggestion from @patlefort on #1106? Is that the proper way to account for WIN32?

**Comment by meshula** [comment]
_WIN32 indeed tests for that platform on both mingw and windows. But the headers exist outside of _WIN32; the test on x86 is in the right direction. Let's settle Linux/win/Apple first with this patch, and then take a step forward to tweak as necessary for mingw.

**Review by meshula — APPROVED** [review]

**Pull request merged**